### ISSUE TITLE
Should string -> int casts be allowed to have underscores?

### ISSUE THREAD

**Issue by ronawho** [chapel-lang/chapel#11756]
https://github.com/chapel-lang/chapel/pull/11464 added support for having `_` as a separator in numeric literals. Should we also support casting numeric strings with underscores to numeric types? 

I think so, but currently they are not allowed:

```chpl
const i = "1_000":int;
writeln(i);
```

```
./castUnderscore
uncaught IllegalArgumentError: bad cast from string '1_000' to int(64)
  castUnderscore.chpl:1: thrown here
  castUnderscore.chpl:1: uncaught here
```

**Associated Future Test(s):**
https://github.com/chapel-lang/chapel/blob/master/test/functions/casts/stringToIntUnderscores.chpl

**Comment by ben-albrecht** [comment]
I would expect (perhaps naively) that any valid numerical literal would also work when casted from string to a numerical type.

**Comment by daviditen** [comment]
Closed with PR #11880.
